app/src/main/kotlin/com/hippo/ehviewer/ui/tools/Modifier.kt

```diff
@@ -1,6 +1,39 @@
 package com.hippo.ehviewer.ui.tools
 
+import androidx.compose.animation.AnimatedVisibilityScope
+import androidx.compose.animation.EnterTransition
+import androidx.compose.animation.ExitTransition
+import androidx.compose.animation.SharedTransitionScope
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
 
 inline fun Modifier.thenIf(condition: Boolean, crossinline block: Modifier.() -> Modifier) =
     if (condition) block() else this
+
+val NoopTransitionsVisibilityScope = TransitionsVisibilityScope(emptySet())
+
+@Stable
+class TransitionsVisibilityScope(val scopes: Set<AnimatedVisibilityScope>)
+
+context(TransitionsVisibilityScope)
+@Composable
+inline fun <T> AnimatedVisibilityScope.advance(block: @Composable TransitionsVisibilityScope.() -> T) = block(remember(scopes, this) { TransitionsVisibilityScope(scopes + this) })
```

Comment:
The `advance` should have a better name

Reply:
Move `AnimatedVisibilityScope` to parameter and name it `with`?

Reply:
SGTM.